x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/support/mapper/ClusterStateRoleMapper.java

```diff
@@ -54,8 +56,8 @@ public final class ClusterStateRoleMapper extends AbstractRoleMapperClearRealmCa
     public ClusterStateRoleMapper(Settings settings, ScriptService scriptService, ClusterService clusterService) {
         this.scriptService = scriptService;
         this.clusterService = clusterService;
-        // this role mapper is disabled by default and only code in other plugins can enable it
-        this.enabled = settings.getAsBoolean(CLUSTER_STATE_ROLE_MAPPINGS_ENABLED, false);
+        // this role mapper is enabled by default and only code in other plugins can disable it
+        this.enabled = settings.getAsBoolean(CLUSTER_STATE_ROLE_MAPPINGS_ENABLED, true);
```

Comment:
(can be done in a followup) I think this is always enabled, so we can remove this setting entirely. We will need to (in order) a) remove from serverless config b) remove from serverless code c) remove here

alternatively, we can register the setting in main code and allow users to disable it, but will also need to test and throw exceptions if the setting is false and K8's defined the configuration (to avoid a write, but never read situation).

I would prefer to just remove it entirely since there isn't a good reason to disable it.